File: src/main/java/com/leijendary/spring/boot/template/document/LocaleDocument.java
package com.leijendary.spring.boot.template.document;

import static org.springframework.data.elasticsearch.annotations.FieldType.Keyword;

import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import lombok.Data;

@Data
public abstract class LocaleDocument {

    @Field(type = Keyword)
    private String language;

    @Field(type = FieldType.Integer)
    private int ordinal;
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/document/LocaleDocument.kt
package com.leijendary.spring.boot.template.core.document

import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType.Integer
import org.springframework.data.elasticsearch.annotations.FieldType.Keyword

abstract class LocaleDocument(
    @Field(type = Keyword)
    var language: String? = null,

    @Field(type = Integer)
    var ordinal: Int = 0
)